src/App.vue

```diff
@@ -95,35 +107,73 @@ export default defineComponent({
         },
         completed(): Todo[] {
             return this.todos.filter(todo => todo.isCompleted);
+        },
+        isAppReady(): boolean {
+            return !this.isLoadingUserSession && !this.isLoadingTodos;
+        },
+        isLoggedIn(): boolean {
+            return Boolean(this.user.uid);
+        },
+        userTodosPath(): string {
+            return `users/${this.user.uid}/todos`;
         }
     },
+    created() {
+        this.setUser();
+    },
     methods: {
+        setUser() {
+            firebaseApp.auth().onAuthStateChanged((user: User) => {
```

Comment:
The linter should tell you that can't type this as `User` 🤔
```suggestion
            firebaseApp.auth().onAuthStateChanged((user: User) => {
```